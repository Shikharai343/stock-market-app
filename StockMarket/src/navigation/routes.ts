export const RouteNames = {
    // Stacks
    app: 'App',
    splash: 'Splash',

    // screens
    home: 'Home',
    stock: 'StockPrice',
};
