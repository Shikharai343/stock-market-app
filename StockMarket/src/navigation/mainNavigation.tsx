import React from 'react';
import { createStackNavigator } from 'react-navigation';
import HomeScreen from '../screens/HomeScreen';
import StockPriceScreen from '../screens/StockPriceScreen';
import {RouteNames} from './routes';

const homeStack = createStackNavigator({
    [RouteNames.home]: HomeScreen,
    [RouteNames.stock]: StockPriceScreen,
});

export const AppNavigator = (homeStack);

AppNavigator.navigationOptions = {
    header: null,
};
