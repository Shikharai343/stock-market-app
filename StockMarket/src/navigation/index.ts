import {
    createAppContainer,
    createSwitchNavigator,
} from 'react-navigation';
import { AppNavigator } from './mainNavigation';
import SplashScreen from '../screens/SplashScreen';
import {RouteNames} from './routes';

export default createAppContainer(createSwitchNavigator(
    {
        [RouteNames.splash]: SplashScreen,
        [RouteNames.app]: AppNavigator,
    },
    {
        initialRouteName: RouteNames.splash,
    },
));
