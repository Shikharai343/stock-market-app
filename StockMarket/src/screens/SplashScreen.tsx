import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {RouteNames} from '../navigation/routes';

interface IProps {
    navigation: any;
}

export default class SplashScreen extends React.Component<IProps, any> {

    public componentDidMount() {
        setTimeout(() => {
            this.props.navigation.navigate(RouteNames.app);
        }, 2000);
    }


    public render() {
        return (
            <View style={styles.container}>
                <Image source={require('../assets/pic2.jpg')}/>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    text: {
        color: 'blue',
        fontSize: 25,
    }
});
