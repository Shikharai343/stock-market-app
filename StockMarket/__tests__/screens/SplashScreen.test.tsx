import React from 'react';
import { shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import SplashScreen from '../../src/screens/SplashScreen';
import {RouteNames} from '../../src/navigation/routes';

jest.useFakeTimers();
let navigation = {
    navigate: jest.fn()
};
let wrapper: any;

describe('Splash Screen testing', () => {
    it('should create a snapshot of the Splash Screen', () => {
        wrapper = shallow(<SplashScreen  navigation={navigation}/>);
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should navigate to app', () => {
        jest.runAllTimers();
        expect(navigation.navigate).toHaveBeenCalledWith(RouteNames.app);
    })
});
