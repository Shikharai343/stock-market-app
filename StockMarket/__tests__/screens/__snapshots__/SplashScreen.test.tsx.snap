// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen testing should create a snapshot of the Splash Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/pic2.jpg",
      }
    }
  />
</View>
`;
